text/0000-update-hook-runs.md

```diff
@@ -0,0 +1,196 @@
+- Start Date: 2017-06-19
+- RFC PR:
+- Yarn Issue:
+
+# Summary
+
+Bring the scripts that run on certain commands closer to parity with NPM.
+
+# Motivation
+
+There are a few changes to the scripts hooks that NPM has made recently, and it
+would be good to bring `yarn` to the same standard.
+
+This means that `yarn` can remove hooks that don't make sense, and stop
+supporting issues brought up by people confused about *why* a certain script is
+running.
+
+Additionally, by tracking closely to what NPM is doing with their hooks, `yarn`
+remains a drop-in replacement, as most people expect it to act.
+
+# Detailed design
+
+### State of Scripts
+
+Below are all the scripts that NPM currently auto-runs, and at which points they
+run. Currently, there is only one deprecation that motivated this RFC, but it is
+anticipated that there will be more in the long run.
+
+This list is kind of long, because I've enumerated all options.
+
+- `prepublish`
+  - before `publish`
+  - before `install`
+    - *Note: this is the deprecation. `prepublish` will no longer run on
+      install*
+    - This hook should *not* be added, as it is removed as of `npm@5`.
+- `prepare`
+  - before `publish`
+  - before `install`
+  - *Note: This is the replacement for the current `prepublish` behaviour.*
+- `prepublishOnly`
+  - before `publish`
+- `prepack`
+  - before `pack`
+  - before `publish`
+- `postpack`
+  - after `pack`
+  - after `publish`
+- `publish`
+  - after `publish`
+    - confusing, but runs *after* the package has been published.
+- `postpublish`
+  - after `publish`
+- `preinstall`
+  - before `install`
+- `install`
+  - after `install`
+    - confusing, but runs *after* the package has been installed.
+- `postinstall`
+  - after `install`
+- `preuninstall`
+  - before `uninstall`
+- `uninstall`
+  - before `uninstall`
+    - confusing, but runs *before* the package is uninstalled
+- `postuninstall`
+  - after `uninstall`
+- `preversion`
+  - before `version`
+- `version`
+  - after `version`
+  - runs *before* the version commit is made
+- `postversion`
+  - after `version`
+  - runs *after* the version commit is made
+- `preshrinkwrap`
+  - before `shrinkwrap`
+- `shrinkwrap`
+  - before `shrinkwrap`
+    - confusing, but runs *before* the shrinkwrap is created
+  - I'm not sure `yarn` needs to include this, given that `shrinkwrap` is not
```

Comment:
the shrinkwrap scripts also run when creating the new `package-lock.json`, so yarn should also run if when creating the `yarn.lock` file, for a closer 1-to-1 match